libnd4j/include/loops/cpu/TrueBroadcastHelper.hpp

```diff
@@ -44,8 +45,24 @@ void TrueBroadcastHelper<X, Y, Z>::exec(const NDArray& xArr, const NDArray& yArr
     const int yRank = yArr.rankOf();
     const int zRank = zArr.rankOf();
 
-    const Nd4jLong zLen  = zArr.lengthOf();
-
+    bool bSpecialCase = (1 == xArr.ews() && 'c' == xArr.ordering() && 1 == yRank);
```

Comment:
Almost. We also must be sure that:
- Y and Z arrays have ews == 1
- Y and Z arrays have 'c' order

Reply:
added